lib/awful/prompt.lua

```diff
@@ -485,16 +485,33 @@ function prompt.run(args, textbox, exe_callback, completion_callback,
                     end
                     if match then
                         local cb
-                        local ret = v[3](command)
+                        local ret, quit = v[3](command)
                         local original_command = command
-                        if ret then
-                            command = ret
+
+                        -- Support both a "simple" and a "complex" way to
+                        -- control if the prompt should quit.
+                        quit = quit == nil and (ret ~= true) or (quit~=false)
+
+                        -- Allow the callback to change the command
+                        command = (ret ~= true) and ret or command
```

Comment:
Sigh, the joy of API compatibility. I don't really understand this and I'm lazy right now, so I'll let @blueyed decide. My only question is: API documentation?